src/_blog/libp2p-and-Ethereum.md

```diff
@@ -0,0 +1,310 @@
+- [ConsenSys (PegaSys)](https://consensys.net/knowledge-base/ethereum-2/teku/) (the team behind [Teku](https://docs.teku.consensys.net/en/latest/) which uses [jvm-libp2p](https://github.com/libp2p/jvm-libp2p))
+- [Status.im](https://status.im/) (the team behind [Nimbus](https://nimbus.team/) which uses [nim-libp2p](https://github.com/status-im/nim-libp2p))
+- [ChainSafe Systems](https://chainsafe.io/) (the team behind [Lodestar](https://lodestar.chainsafe.io/) which uses [js-libp2p](https://github.com/libp2p/js-libp2p))
+- [Ethereum Foundation](https://ethereum.foundation/)
+
+During this long developmental cycle, libp2p was running in the above validators on the Beacon Chain.
+And once the Beacon Chain was merged with the original execution, libp2p was fully integrated into the Ethereum mainnet!
+
+# How libp2p is used by Ethereum validators 🔍
+
+So which exact libp2p modules are utilized by validators?
+Per the specification, Ethereum validators use the following libp2p features:
+- [Transport](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#transport): TCP
+- [Encryption](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#encryption-and-identification): [Noise](https://github.com/libp2p/specs/tree/master/noise)
+- [Protocol Negotiation](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#protocol-negotiation): [Multistream Select 1.0](https://github.com/multiformats/multistream-select/#protocol)
+- [Stream Multiplexing](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#multiplexing): [mplex](https://github.com/libp2p/specs/tree/master/mplex) or [yamux](https://github.com/hashicorp/yamux/blob/master/spec.md)
+- [Messaging](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#the-gossip-domain-gossipsub): [GossipSub v1.1](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md)
+
+All of the modules above aside from yamux are mandatory.
+One notable omission is the [libp2p Kademlia DHT](https://github.com/libp2p/specs/tree/master/kad-dht); The networking specification instead specifies the use of discv5.
+The rationale is [provided here](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#why-are-we-using-discv5-and-not-libp2p-kademlia-dht).
+
+The specific design details are out of scope for this post but are [described in great detail here](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#design-decision-rationale).
+Please check it out to see why TCP was chosen (and not QUIC), why Noise, why GossipSub etc.
+
+## Network Statistics 📊
+
+Currently there are over [495,000 validators](https://beaconcha.in/validators) with this [current client diversity](https://clientdiversity.org/#distribution).
+libp2p is running on all those consensus clients; an amazing testament to 6+ years of hard work!
```

Comment:
They're not all online though, are they? Do we have an estimate for the actual size of the network?

Reply:
Those validators are listed as active so I would say yes.
Regarding network size i.e. consensus clients actually running libp2p, the number is around 10,000: https://www.nodewatch.io/ 
That info seems to be fetched using a devp2p crawler https://github.com/ChainSafe/nodewatch-api

Reply:
Past discussion on the number of Ethereum nodes running libp2p:

https://filecoinproject.slack.com/archives/C03K82MU486/p1668620946724079

I find it not directly obvious from the above, that it is 10_000 nodes running libp2p and not 495_000 nodes. Do you agree? What do you think of rephrasing the above?

Reply:
I see thanks, from that slack thread it looks like ChainSafe's nodewatch.io dashboard is also indexing testnets and reports ~11k nodes now.
In comparison, Miga Labs' dashboard indicates there are about ~5k beacon chain nodes. So it seems 5k is the more accurate number.

I've rephrased the paragraph too.